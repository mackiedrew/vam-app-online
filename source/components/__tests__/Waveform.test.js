import Subject from '../Waveform'

const mockProps = {
  blocks: [
    {
      start: 0,
      end: 3,
      amplitude: 4,
    },
  ],
  maxAmplitude: 10,
}

describe('<Waveform /> structure', () => {

  it('renders without crashing with no grains', () => {
    shallow(<Subject />)
  })

  it('renders without crashing with some grains', () => {
    shallow(<Subject {...mockProps} />)
  })

  it('renders as a div with class: `waveform`', () => {
    const wrapper = shallow(<Subject />)
    expect(wrapper.is('div.waveform')).toEqual(true)
  })

  it('renders with no <WaveBlock />s when no blocks are supplied', () => {
    const wrapper = shallow(<Subject />)
    expect(wrapper.find('WaveBlock')).toHaveLength(0)
  })

  it('renders with a single <WaveBlock> when provided with one block', () => {
    const wrapper = shallow(<Subject {...mockProps} />)
    expect(wrapper.find('WaveBlock')).toHaveLength(1)
  })

})