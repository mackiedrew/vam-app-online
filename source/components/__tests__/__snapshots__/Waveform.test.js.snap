// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Waveform /> renders correctly with a grains 1`] = `
<div
  className="waveform"
>
  <button
    className="wave-block enabled"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(240, 240, 240)",
        "flexGrow": 3,
      }
    }
  >
    <div
      className="amplitude"
      style={
        Object {
          "height": "40%",
          "opacity": "0.4",
        }
      }
    />
  </button>
</div>
`;

exports[`<Waveform /> renders correctly with empty grains 1`] = `
<div
  className="waveform"
/>
`;
